### PR TITLE
Update JWKS provider for full JWKS Uri, and optional JWK Uri

### PR DESCRIPTION
**Pull request by khorwood**
As per discussion with @j3parker

- Will need to migrate Auth Service `public_key_endpoint` to:
  - jwks_endpoint: `x => x + /.well-known/jwks`
  - jwk_endpoint: `x => x + /jwk/`

### CODE REVIEW

File: src/D2L.Security.OAuth2/Keys/Default/Data/JwksProvider.cs [hunk 1]
```diff
@@ -9,36 +9,41 @@
...
 		async Task<JsonWebKeySet> IJwksProvider.RequestJwkAsync( string keyId ) {
-			var url = GetJwkEndpoint( m_authEndpoint, keyId );
+			var url = GetJwkEndpoint( m_jwkEndpoint, keyId );
 			try {
+				if( url == null ) {
+					return await ( this as IJwksProvider ).RequestJwksAsync().SafeAsync();
+				}
+
 				using( var res = await m_httpClient.GetAsync( url ).SafeAsync() ) {
 					// This is temporary while we try to fully deprecate the
```
Comment (j3parker) [inline src/D2L.Security.OAuth2/Keys/Default/Data/JwksProvider.cs]: TODO for me: re-remove this as in #110

File: src/D2L.Security.OAuth2/Keys/Default/Data/JwksProvider.cs [hunk 2]
```diff
@@ -9,36 +9,41 @@
...
 			}
 		}
 
 		async Task<JsonWebKeySet> IJwksProvider.RequestJwkAsync( string keyId ) {
-			var url = GetJwkEndpoint( m_authEndpoint, keyId );
+			var url = GetJwkEndpoint( m_jwkEndpoint, keyId );
 			try {
+				if( url == null ) {
+					return await ( this as IJwksProvider ).RequestJwksAsync().SafeAsync();
+				}
```
Comment (j3parker) [inline src/D2L.Security.OAuth2/Keys/Default/Data/JwksProvider.cs]: Does this need to be in the `try`? It looks like that method catches the same exceptions.

File: src/D2L.Security.OAuth2/Keys/Default/Data/JwksProvider.cs [hunk 3]
```diff
@@ -70,7 +75,7 @@ Uri authEndpoint
 			}
 		}
 
-		string IJwksProvider.Namespace => m_authEndpoint.AbsoluteUri;
+		string IJwksProvider.Namespace => m_jwksEndpoint.AbsoluteUri;
```
Comment (j3parker) [inline src/D2L.Security.OAuth2/Keys/Default/Data/JwksProvider.cs]: @omsmith FYI. @khorwood and I discussed this in person. There are some implicit assumptions here but seems reasonable?

**Comment by j3parker** [comment]
Add some test coverage for that?
That'd be fine as a follow-up.

File: src/D2L.Security.OAuth2/Keys/Default/Data/JwksProvider.cs (hunk 2 above)
Comment (khorwood) [reply src/D2L.Security.OAuth2/Keys/Default/Data/JwksProvider.cs]: Probably not, no.

Comment (khorwood) [reply src/D2L.Security.OAuth2/Keys/Default/Data/JwksProvider.cs]: - [X] move block out of try-catch

**Comment by khorwood** [comment]
@j3parker Added some extra tests and moved the fallback outside of the try-catch.

**Pull request merged**